File: java/v1/src/main/java/com/api/v1/controllers/creditcard/CreditCardRegistrationController.java
package com.api.v1.controllers.creditcard;

import com.api.v1.dtos.creditcard.CreditCardRegistrationRequestDto;
import com.api.v1.dtos.creditcard.CreditCardResponseDto;
import com.api.v1.services.customer.CreditCardRegistrationService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/credit-cards")
public class CreditCardRegistrationController {

    @Autowired
    private CreditCardRegistrationService service;

    @PostMapping
    @ResponseStatus(value = HttpStatus.CREATED)
    public Mono<CreditCardResponseDto> register(@Valid @RequestBody CreditCardRegistrationRequestDto requestDto) {
        return service.register(requestDto);
    }

}

File: v2/src/main/kotlin/com/api/v2/creditcard/controllers/CreditCardRegistrationController.kt
package com.api.v2.creditcard.controllers

import com.api.v2.creditcard.services.CreditCardRegistrationService
import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.creditcard.dtos.CreditCardResponseDto
import jakarta.validation.Valid
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v1/credit-cards")
class CreditCardRegistrationController {

    @Autowired
    private lateinit var service: CreditCardRegistrationService

    @PostMapping
    @ResponseStatus(value = HttpStatus.CREATED)
    suspend fun register(@RequestBody creditCard: @Valid CreditCard): CreditCardResponseDto {
        return service.register(creditCard)
    }

}